mautrix_twitter/user.py

```diff
@@ -333,6 +334,15 @@ async def get_direct_chats(self) -> dict[UserID, list[RoomID]]:
             if portal.mxid
         }
 
+    async def get_portal_with(self, puppet: pu.Puppet, create: bool = True) -> po.Portal | None:
+        if not self.twid:
+            return None
+        return await po.Portal.get_by_twid(
+            puppet.twid,
+            receiver=self.twid,
+            conv_type=ConversationType.ONE_TO_ONE if create else None,
+        )
```

Comment:
I think twitter also requires DMs to be created with a separate API call, the conversation ID is `$creator-$recipient` rather than just the other party ID.

```suggestion
        return None
```